Diabetes/projects/diabetes_project_1.html

```diff
@@ -0,0 +1,87 @@
+<html>
+
+<head>
+  <script src='https://ajax.googleapis.com/ajax/libs/jquery/3.4.1/jquery.min.js'></script>
+  <script type='text/javascript'>
+    $(document).ready(function () {
+      $('body').append(`<p id="prediction">Prediction: <p>`);
+
+      $.fn.serializeObject = function () {
+        var object = {};
+        $.each(this.serializeArray(), function () {
+          if (object[this.name] !== undefined) {
+            [object[this.name]].push(this.value || '');
+          } else {
+            object[this.name] = this.value || '';
+          }
+        });
+        return object;
+      };
+
+      var DATA;
+      function predict(data = DATA) {
+        console.log('data', DATA);
+        var ENDPOINT = $('[name=url]').val();
+        return fetch(ENDPOINT, {
+          method: 'POST',
+          body: data,
+        })
+          .then((res) => res.json())
+          .then((resJSON) => {
+            console.log('resJSON', resJSON);
+            resJSON = JSON.parse(resJSON['body']);
+            var innerHTML = $('#prediction').filter('p').text();
+            var label = innerHTML.substring(innerHTML.indexOf('Prediction:') + 'Prediction:'.length);
+            $('#prediction').html($('#prediction').html().replace(label, resJSON['predicted_label']));
+          })
+          .catch(err => console.log('err', err));
+      };
+
+      $('#PredictButton').click(function (data = DATA) {
+        DATA = JSON.stringify($('form').serializeObject());
+        predict();
```

Comment:
then here, you can chain the function to make the updates to the dom.

```
predict().then(body => updateDom(body))
```

Reply:
also you are not passing the DATA from the form into the predict function. make sure DATA is a string before passing into the predict function